File: leetcode.com/0713_subarray_product_less_than_k/solution.c
int numSubarrayProductLessThanK(int *nums, int numsSize, int k) {
    int total = 0;

    for (int i = 0 ; i < numsSize ; i++) {
        int p = 1;
        for (int j = i ; j < numsSize ; j++) {
            if (p*nums[j] >= k)
                break;
            p *= nums[j];
            total++;
        }
    }

    return total;
}

File: leetcode.com/0713_subarray_product_less_than_k/solution.go
func numSubarrayProductLessThanK(nums []int, k int) int {
    t := 0

    for i := 0 ; i < len(nums) ; i++ {
        p := 1
        for j := i ; j < len(nums) ; j++ {
            if (nums[j]*p < k) {
                p *= nums[j]
                t += 1
            } else {
                 break
            }
        }
    }

    return t
}
